synapse/storage/databases/main/registration.py

```diff
@@ -1760,6 +1760,260 @@ def start_or_continue_validation_session_txn(txn):
             start_or_continue_validation_session_txn,
         )
 
+    async def registration_token_is_valid(self, token: str) -> bool:
```

Comment:
Note that these methods being in `RegistrationStore` rather than `RegistrationWorkerStore` will mean that worker processes cannot call them. Since `^/_matrix/client/(r0|unstable)/register$` is already supported by workers - we should make sure that this change doesn't drop that support.

All that's necessary should be just moving these methods into `RegistrationWorkerStore` instead - which worker processes make use of, rather than `RegistrationStore` (which inherits from `RegistrationWorkerStore` and is used by the main process only).

Reply:
Ok, I've moved them but haven't tested with workers at all

Reply:
That's alright. That would be done in Complement CI sometime in the future when Complement worker infrastructure is up and running.

Reply:
Tested with a generic worker and it works :-)